benchmarks/symbol_to_proc_vs_block.rb

```diff
@@ -0,0 +1,333 @@
+# We have not decided how we are going to handle the symbol to proc vs
+# block implementations. This is because the block is faster on JRuby
+# but slower on Ruby 1.9.2 through 2.2.2.
+
+require 'benchmark'
+
+RANGE = (1..100)
+
+def sym_to_proc_with_map
+  RANGE.map(&:to_s)
+end
+
+def sym_to_proc_with_each
+  RANGE.each(&:to_s)
+end
+
+def block_with_map
+  RANGE.map { |i| i.to_s }
+end
+
+def block_with_each
+  RANGE.each { |i| i.to_s }
+end
+
+begin
+  require 'benchmark/ips'
+
+  [10, 100, 1000, 10_000].each do |size|
+    range = (1..size)
+
+    puts "\nUsing Array#map size #{size}:"
+    Benchmark.ips do |x|
+      x.report('Block         ') { range.map { |i| i.to_s } }
```

Comment:
Benchmark is incorrect. It should be:

```suggestion
      x.report('Block         ') { block_with_map }
```

This is the same for all the others.